app/src/main/java/com/example/cinemapp/ui/main/DetailsFragment.kt

```diff
@@ -0,0 +1,159 @@
+        _binding = FragmentDetailsBinding.inflate(inflater, container, false)
+        binding.clContent.visibility = View.INVISIBLE
+        viewModel.getMovieDetails(args.movieId)
+        setupAdapter()
+        return binding.root
+    }
+
+    override fun onStart() {
+        super.onStart()
+        observeFlowSafely(viewModel.state) {
+            it.details?.let { details ->
+                setupView(details)
+                binding.clContent.visibility = View.VISIBLE
+                binding.cpiLoading.visibility = View.GONE
+            }
+            castAdapter.setCast(it.cast)
+            mediaAdapter.setMedia(it.media)
+        }
+    }
+
+    private fun setupView(details: MovieDetails) {
+        with(binding) {
+            tbTop.setNavigationIcon(R.drawable.vic_arrow_back)
+            tbTop.setNavigationOnClickListener {
+                requireActivity().onBackPressedDispatcher.onBackPressed()
+            }
+
+            tvTitle.text = details.title
+            tvGenres.text = details.genres.joinToString(", ") { genre -> genre.name }
+            val runtime =
+                "${(details.runtime / 60)}h ${details.runtime % 60}min"
+            tvRuntime.text = runtime
+
+            if (details.overview.length > OVERVIEW_MAX_CHARACTERS) {
+                val spannableString = makeOverview(details.overview) {
+                    tvOverview.text = details.overview
+                }
+                tvOverview.movementMethod = LinkMovementMethod.getInstance()
+                tvOverview.setText(spannableString, TextView.BufferType.SPANNABLE)
+            } else tvOverview.text = details.overview
+
+            ivArrowLeft.setOnClickListener {
+                moveRecyclerView(-1)
+            }
+            ivArrowRight.setOnClickListener {
+                moveRecyclerView(1)
+            }
+        }
+    }
+
+    private fun setupAdapter() {
+        with(binding) {
+            rvCast.adapter = castAdapter
+            rvCast.layoutManager =
+                LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+
+            rvMedia.adapter = mediaAdapter
+            rvMedia.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+            PagerSnapHelper().attachToRecyclerView(rvMedia)
+        }
+    }
+
+    private fun moveRecyclerView(delta: Int) {
+        with(binding) {
+            val lm = rvMedia.layoutManager as LinearLayoutManager
+            val smoothScroller = object : LinearSmoothScroller(context) {
+                override fun getHorizontalSnapPreference(): Int {
+                    return SNAP_TO_START
+                }
+            }
+            smoothScroller.targetPosition = MathUtils.clamp(
+                lm.findFirstCompletelyVisibleItemPosition() + delta,
+                0,
+                lm.itemCount - 1
+            )
+
+            lm.startSmoothScroll(smoothScroller)
+        }
+    }
+
+    private fun makeOverview(
```

Comment:
This seems like some kind of utility, right? 😄 

So, it could be a good thing to extract this method into separate file, called something like `TextUtil.kt